db/change_cache_test.go

```diff
@@ -1854,8 +1855,11 @@ func TestMaxChannelCacheConfig(t *testing.T) {
 	for _, val := range channelCacheMaxChannels {
 		options := CacheOptions{
 			ChannelCacheOptions: ChannelCacheOptions{
-				MaxNumChannels: val,
+				MaxNumChannels:  val,
+				ChannelCacheAge: DefaultChannelCacheAge,
 			},
+			CachePendingSeqMaxWait: DefaultCachePendingSeqMaxWait,
+			CacheSkippedSeqMaxWait: DefaultSkippedSeqMaxWait,
```

Comment:
I'm not sure why these default option values are being explicitly added. The code in `newChannelCacheWithOptions` should be setting default values for anything that isn't explicitly set?

Reply:
The code in newChannelCacheWithOptions is not being used in the test and the CacheOptions are mocked without any value for CachePendingSeqMaxWait, CacheSkippedSeqMaxWait options instead. This makes the Ticker in panic whilst launching the background tasks due to 0 value for the duration. I felt this change can go along with the CBG-702.

Reply:
Revised NewBackgroundTask to return an error if the task interval is less than or equal to 0. A BackgroundTaskError will be returned instead of creating the background task if the interval is illegal. Also modified the channel cache related test and code base to handle this.